src/main/java/com/netgrif/application/engine/petrinet/domain/Component.java

```diff
@@ -40,4 +41,13 @@ public Component(String name, Map<String, String> properties, List<Icon> optionI
         this.properties = properties;
         this.optionIcons = optionIcons;
     }
+
+    @Override
+    public Component clone() {
+        Component clone = new Component();
```

Comment:
please re-do via the constructor:
 https://www.artima.com/articles/josh-bloch-on-design#part13

Reply:
done